conformance/utils/suite/suite.go

```diff
@@ -62,6 +62,11 @@ type Options struct {
 	BaseManifests    string
 	MeshManifests    string
 	NamespaceLabels  map[string]string
+
+	// MeshNamespaceLabels are implentation-specific labels that should be added
+	// to namespaces where mesh-enabled services will be deployed.
+	MeshNamespaceLabels map[string]string
```

Comment:
Why do we need this when we already have `NamespaceLabels`? It only applies to "mesh workload" namespaces not everywhere?

Reply:
>  It only applies to "mesh workload" namespaces not everywhere?

Yes, that is the difference.

It wasn't clear to me what the intended use of `NamespaceLabels` is, so I added another field. What kind of labels would we want to add to ALL namespaces, mesh or not?

Reply:
@shaneutt, yes, that is my assumption. Mesh egress tests, for example.

Reply:
Actually the more general problem is that if one wants to run both the Gateway and Mesh tests, there is no way to label only the mesh namespace(s), but not the ones hosting plain k8s services that the Gateway/Route tests use. Maybe that's what we want? When running Mesh and Gateway tests together, the Gateway tests switch to mesh-ingress (instead of k8s-ingress) tests?

Reply:
@frankbu This seems like something that's maybe a bit too implementation-specific to include in our conformance test helpers unless I'm missing something. Would it be possible to just have mesh tests runs separately from ingress tests?

Reply:
@robscott, yes, absolutely, I think that would be fine. The only potential issue is if we have mesh tests that want to call services outside of the mesh (egress testing). In Istio, we use a "non-mesh" namespace to run services that simulate external services for this kind of testing.

Reply:
@robscott @shaneutt I removed `MeshNamespaceLabels` and just use the existing `NamespaceLabels` for now. If in the future we need to differentiate mesh/not-mesh namespaces we can figure out a way to do it then.

So now this PR is only adding a flag to set `NamespaceLabels` and fixing up pod.go to properly check that the version routing test is actually calling the expected backends.